service/controller/v14/resource/pod/delete.go

```diff
@@ -83,20 +84,77 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 			r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for inspection of the reconciled pod")
 		}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "inspecting drainer config for the guest cluster")
+		if drainerConfig.Status.HasDrainedCondition() {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config of guest cluster has drained condition")
 
-		if !drainerConfig.Status.HasFinalCondition() {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config of guest cluster has no final state")
-			resourcecanceledcontext.SetCanceled(ctx)
-			finalizerskeptcontext.SetKept(ctx)
-			r.logger.LogCtx(ctx, "debug", "canceling reconciliation for pod")
+			err := r.finishDraining(ctx, currentPod, drainerConfig)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
 
-			return nil
+		if drainerConfig.Status.HasTimeoutCondition() {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config of guest cluster has timeout condition")
+
+			err := r.finishDraining(ctx, currentPod, drainerConfig)
+			if err != nil {
+				return microerror.Mask(err)
+			}
 		}
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config of guest cluster has no drained condition")
+	resourcecanceledcontext.SetCanceled(ctx)
+	finalizerskeptcontext.SetKept(ctx)
+	r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation for pod")
+
+	return nil
+}
+
+func (r *Resource) createDrainerConfig(ctx context.Context, pod *corev1.Pod) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "creating drainer config for guest cluster node")
+
+	apiEndpoint, err := key.ClusterAPIEndpointFromPod(pod)
+	if err != nil {
+		return microerror.Mask(err)
+	}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "drainer config of guest cluster has final state")
+	n := pod.GetNamespace()
+	c := &corev1alpha1.DrainerConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pod.GetName(),
+		},
+		Spec: corev1alpha1.DrainerConfigSpec{
+			Guest: corev1alpha1.DrainerConfigSpecGuest{
+				Cluster: corev1alpha1.DrainerConfigSpecGuestCluster{
+					API: corev1alpha1.DrainerConfigSpecGuestClusterAPI{
+						Endpoint: apiEndpoint,
+					},
+					ID: pod.GetNamespace(),
+				},
+				Node: corev1alpha1.DrainerConfigSpecGuestNode{
+					Name: pod.GetName(),
+				},
+			},
+			VersionBundle: corev1alpha1.DrainerConfigSpecVersionBundle{
+				Version: "0.1.0",
```

Comment:
Something I just thought about:
How do we plan to manage this? What happens if the version would move up, how do we remember to change it here?

Reply:
I am not sure. This is definitely something we should address. How is the version dispatching managed with `chart-operator`? Don't we have the same issue there that some version bundle versions have to be forwarded?

Reply:
Yes we do, but we don't address it either.
I tried to get at this in slack: https://gigantic.slack.com/archives/C2MS2VB37/p1531472745000275
This is quite concerning to me, it is bound to the `Versioning of Operators` story in my opinion but it wasn't on my radar at all.

Reply:
Lets make sure we track this somewhere. Addressing it with the release operator makes sense. If there is need lets get together and have a brainstorm.